plugin/evm/block_builder.go

```diff
@@ -58,10 +59,11 @@ func (vm *VM) NewBlockBuilder(extraMempool extension.BuilderMempool) *blockBuild
 }
 
 // handleGenerateBlock is called from the VM immediately after BuildBlock.
-func (b *blockBuilder) handleGenerateBlock() {
+func (b *blockBuilder) handleGenerateBlock(currentHeight uint64) {
```

Comment:
I think we should be flagging on the parent hash rather than the height. There are some weird potential reorgs where we successfully build a block, but then get reorged and then attempt to build a new block at the same height (with no incorrect behavior).